tools/build_defs/cc_import.bzl

```diff
@@ -0,0 +1,208 @@
+            library = library,
+        )
+
+def _build_libraries_to_link_and_runtime_artifact(ctx, files, cc_toolchain, targets_windows):
+    static_library = None
+    if files.static_library != None:
+        static_library = _build_static_library_to_link(ctx, files.static_library)
+
+    shared_library = None
+    runtime_artifact = None
+    if files.shared_library != None:
+        shared_library = _build_shared_library_to_link(ctx, files.shared_library, cc_toolchain, targets_windows)
+        runtime_artifact = shared_library.artifact()
+
+    interface_library = None
+    if files.interface_library != None:
+        interface_library = _build_interface_library_to_link(ctx, files.interface_library, cc_toolchain, targets_windows)
+
+    dynamic_library_for_linking = None
+    if interface_library != None:
+        dynamic_library_for_linking = interface_library
+    else:
+        dynamic_library_for_linking = shared_library
+
+    return (static_library, dynamic_library_for_linking, runtime_artifact)
+
+def _build_cc_link_params(
+        ctx,
+        static_library,
+        dynamic_library,
+        runtime_artifact):
+    static_shared = None
+    static_no_shared = None
+    if static_library != None:
+        static_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(static_library),
+        )
+        static_no_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(static_library),
+        )
+    else:
+        static_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(dynamic_library),
+            dynamic_libraries_for_runtime = _to_depset(runtime_artifact),
+        )
+        static_no_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(dynamic_library),
+            dynamic_libraries_for_runtime = _to_depset(runtime_artifact),
+        )
+
+    no_static_shared = None
+    no_static_no_shared = None
+    if dynamic_library != None:
+        no_static_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(dynamic_library),
+            dynamic_libraries_for_runtime = _to_depset(runtime_artifact),
+        )
+        no_static_no_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(dynamic_library),
+            dynamic_libraries_for_runtime = _to_depset(runtime_artifact),
+        )
+    else:
+        no_static_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(static_library),
+        )
+        no_static_no_shared = cc_common.create_cc_link_params(
+            ctx = ctx,
+            libraries_to_link = _to_depset(static_library),
+        )
+
+    return (static_shared, static_no_shared, no_static_shared, no_static_no_shared)
```

Comment:
Instead of returning a tuple, why not return a struct? Then you will not ever make a mistake of what goes where in the tuple